Some suggestions for your code

MarkusLange opened the pull request:
Hi Bercik,

nice that you take the work of Kerwood future, I have overlooked it and want to make you some suggestions for the code for you to you service a little bit easier:

- allways use of the latest ruTorrent
- no need to care about the distro version
- remove unnecessary stuff
- add support for arm* systems (scgi)
- correcting external ipv4
- deactivate not supportet plugins
- redirect http to https
- correct terminal colors
- fix changelog and todo view
- and a little bit there and there

Take a look and grep what you like

BR
Markus

MarkusLange committed:
Massiv cleaning

cleaning and sorting the code for a better plain view, make FASTMODE standard, and download Changelog and TODO to show them if selected on the menu, also remove all files after quit

MarkusLange committed:
correct leftover misc

correkt some leftover misc after cleaning

MarkusLange committed:
Remove FASTMODE

remove FASTMODE from source

MarkusLange committed:
moving some lines around

move some lines around

MarkusLange committed:
Update README.md

MarkusLange committed:
Update Rt-Install-minimal

Clean code
auto use of latest ruTorrent
auto install scgi from external source if needed
made script debian-/derivate indipendent
remove plugin handling
correcting external IP to IPV4

MarkusLange committed:
Update Rt-Install-minimal

clean code
remove demo-mode stuff
remove display help
remove distro detection
remove not needed ports.conf
add https redirection
correct colors
make webuser password look nice
add armel support
put deactivate plugins routine in loop

MarkusLange committed:
Update Rt-Install-minimal

remove unused variables
remove change http port, not needed since http to https redirection
edit rtorrent port-range function to show actuall range and add save by empthy range
shift PREPARE_CONFIG_FILES around to repair Change rTorrent Port-Range

MarkusLange committed:
Update Rt-Install-minimal

show TODO and Changelog view correct for terminal
show port range correktly inside MENU

MarkusLange committed:
Update Rt-Install-minimal

some code cleaning
shifting some lines

Bercik1337 commented:
I see you desperately want to remove FASTMODE everywhere. Why? It's very useful for r&d stuff to quickly see if script is working properly.
Leaving crossed off unsupported systems is also a good thing. People tend to use EOL system and then whine it's not working. Now it's clear for them - it won't fly

Same goes for DEMO mode. When I shoot video, it saves me a lot of time in editing afterwards. 

Arm scgi section looks nice, and disabling plugins section as well. 

Extra hashes in Heavy I/O configuration are there for a reason.

It seems your PR is work of many people. Once "FAIL" is renamed to "fail" and others "fail" to "failed".

All and all, I think I will pull in few lines. Probably in 2 weeks.

MarkusLange commented:
> I see you desperately want to remove FASTMODE everywhere. Why? It's very useful for r&d stuff to quickly see if script is working properly.

For this I have changed `clear` to `clear -x` so you get a new screen but everythink is visible above with scrolling
Another point is, the script takes some time so why punish the user with more time to wait.
> Leaving crossed off unsupported systems is also a good thing. People tend to use EOL system and then whine it's not working. Now it's clear for them - it won't fly

Maybe who is the one to judge, I get it working on Debian 9 without any problems, so EOL is relativ,
and every time there is a new derivate version or one hits EOL the script need to be altered 
> Same goes for DEMO mode. When I shoot video, it saves me a lot of time in editing afterwards.

So this case is a usecase for you, for parts like this screenshorts or videos I would prefer editing this script for the special needs not to integrate it in a version for everyone
> Arm scgi section looks nice, and disabling plugins section as well.
> 
> Extra hashes in Heavy I/O configuration are there for a reason.

if you mean the `Rutracker_check` Plugin this is not working based on the folder rights since home moved to 750, others, like www-date are prevent to use it
Or did you mean the rtorrent.rc file I changed nothing I just remove some `#` in the lines to make it look more even
> It seems your PR is work of many people. Once "FAIL" is renamed to "fail" and others "fail" to "failed".

May the many have the same ideas then maybe, I renamed it because of reasons, everything is lowcase why this not?
> All and all, I think I will pull in few lines. Probably in 2 weeks.

Bercik1337 commented:
> > Arm scgi section looks nice, and disabling plugins section as well.
> > Extra hashes in Heavy I/O configuration are there for a reason.
> 
> if you mean the `Rutracker_check` Plugin this is not working based on the folder rights since home moved to 750, others, like www-date are prevent to use it Or did you mean the rtorrent.rc file I changed nothing I just remove some `#` in the lines to make it look more even

Sorry maybe we misunderstood each other. Heavy I/O is placed in `rtorrent.rc`. Reasoning for extra hashes is: this is "extra" feature, not a must-have for everyone. I wanted to "separate" it somehow from the rest of the code. It actually surprised me there are people with  1Gbit+ pipe who use my script. This section is for them :)


As for scgi and plugin disable, it's great! I'm definitely going to pull that into my script. 10/10


This will be a challenge for me, because I only had one PR *in my life*, and this time - I would like to dissect some lines from your proposal. Because some I will surely pull in, and other I would stop. So just letting you know, it will take time :)

MarkusLange committed:
Update Rt-Install-minimal

add a discription what comes next
expand some "----" lines
rename apache rutorrent .conf
add some HEADER lines
remove SET_HTTP_PORT complete

MarkusLange commented:
> > > Arm scgi section looks nice, and disabling plugins section as well.
> > > Extra hashes in Heavy I/O configuration are there for a reason.
> > 
> > 
> > if you mean the `Rutracker_check` Plugin this is not working based on the folder rights since home moved to 750, others, like www-date are prevent to use it Or did you mean the rtorrent.rc file I changed nothing I just remove some `#` in the lines to make it look more even
> 
> Sorry maybe we misunderstood each other. Heavy I/O is placed in `rtorrent.rc`. Reasoning for extra hashes is: this is "extra" feature, not a must-have for everyone. I wanted to "separate" it somehow from the rest of the code. It actually surprised me there are people with 1Gbit+ pipe who use my script. This section is for them :)
> 
> As for scgi and plugin disable, it's great! I'm definitely going to pull that into my script. 10/10
> 
> This will be a challenge for me, because I only had one PR _in my life_, and this time - I would like to dissect some lines from your proposal. Because some I will surely pull in, and other I would stop. So just letting you know, it will take time :)

Use what you like, and take your time, no rush, and if you have questions about my code what it did exactly ask

MarkusLange committed:
Update Rt-Install-minimal

expand the comment for thr Plugin deactivation and add geoip
add some color inside INSTALL_COMPLETE window

MarkusLange committed:
Update Rt-Install-minimal

lengthen some lines
make creating self-signed certificate look nice
shifting some lines around

MarkusLange committed:
Update Rt-Install-minimal

move a2enmod ssll & rewrite
reintroduce os detection with an smaller footprint

MarkusLange committed:
Update Rt-Install-minimal

correct space to tab

MarkusLange commented:
Bercik I add a different version of DETECTOS with a smaller footprint so there a less lines to change if any release or end of life is happen

MarkusLange committed:
Update Rt-Install-minimal

fix to match exact match by detect os

MarkusLange committed:
Update Rt-Install-minimal

MarkusLange committed:
Update README.md

MarkusLange committed:
Massive changes

    * allways use of the latest ruTorrent
    * no need to care about the distro version
    * remove unnecessary stuff
    * add support for arm* systems (scgi)
    * correcting external ipv4
    * deactivate not supportet plugins
    * redirect http to https
    * correct terminal colors
    * add autodl-irssi plugins
    * update .rtorrent.rc to the new commands
    * fix changelog and todo view
    * and a little bit there and there

MarkusLange committed:
little corrections

MarkusLange committed:
some little fixes

a little bit of shifting around

MarkusLange committed:
some fixes

* put functions together for more order
* make pre-installation packages fully silent
* correct autoredirect from http to https

MarkusLange committed:
Update README.md

MarkusLange committed:
add ngnix and lighttpd as webserver to the script

	* now choose between apache2, nginx and lighttpd as webserver (Rt-Install-minimal-apache2_ngnix_lighttpd.bash)
	* create htaccess passwords now with openssl
	* remove ToDo-List from the Menu

MarkusLange committed:
Update README.md

MarkusLange committed:
rewrite to rpc.socket

rewrite to user rpc.socket
add ruTorrent update
move rtorrent user and rutorrent user input to the menu

MarkusLange committed:
Update README.md

MarkusLange committed:
Update README.md

MarkusLange committed:
Update README.md

MarkusLange committed:
eddit some parts

cleaning the code

MarkusLange committed:
Update Rt-Install-minimal-apache2_ngnix_lighttpd_prc-socket.bash

MarkusLange committed:
quick update

MarkusLange committed:
Remove unused function

Remove function WAIT_A_MINUTE

shifting a bit around

MarkusLange committed:
Update README.md

MarkusLange committed:
Merge branch 'master' of https://github.com/MarkusLange/rt-auto-install

MarkusLange committed:
fix missing php-xml

fix RSS Plugin install missing php-xml for dom

MarkusLange committed:
Update to Version 3.0

deactivate Plugin dump

MarkusLange committed:
Update to V2.5

* adopt the actual ruTorrent branches (v4 & v5) since autodl is
  brocken in v5
* transfer naming from Bercik to me for less confusion
* redirect Todo and Changelog

MarkusLange committed:
refactor To-Do and Changelog

refactor To-Do and Changelog

MarkusLange committed:
Refactor ToDo List

again

MarkusLange committed:
Rename TODO to ToDo

MarkusLange committed:
Update Rt-Install-minimal-apache2_ngnix_lighttpd_prc-socket_choose_branche.bash

MarkusLange committed:
small typo fix

small typo fix inside the ToDo list

MarkusLange committed:
adopt new ruTorrent config.php

adopt new ruTorrent config.php

MarkusLange committed:
Update Rt-Install-minimal-apache2_ngnix_lighttpd_prc-socket_choose_branche.bash

MarkusLange committed:
VERSION 1.7   2025-17-11

* Added lets encrypt support
* Fix some functions missing double bracket
* Fixed ruTorrent versions read out from github

MarkusLange committed:
Update README.md
